EncoreTickets.SDK/Utilities/DataStructures/Tree/ReadOnlyTree.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EncoreTickets.SDK.Utilities.BaseTypesExtensions;
+
+namespace EncoreTickets.SDK.Utilities.DataStructures.Tree
+{
+    public class ReadOnlyTree<TKey, TValue> : ITree<TKey, TValue>
+    {
+        private ReadOnlyTree()
+        {
+            Children = new LinkedList<ReadOnlyTree<TKey, TValue>>();
+        }
+
+        public TKey Key { get; private set; }
+
+        public TValue Item { get; private set; }
+
+        private LinkedList<ReadOnlyTree<TKey, TValue>> Children { get; }
+
+        private ReadOnlyTree<TKey, TValue> Parent { get; set; }
+
+        public static ReadOnlyTree<TKey, TValue> BuildOne(IEnumerable<TValue> source, Func<TValue, TKey> keySelector, Func<TValue, TKey> parentKeySelector)
+        {
+            return BuildMany(source, keySelector, parentKeySelector).FirstOrDefault();
+        }
```

Comment:
Why do you need to build all the items to pick one? Could you rewrite this to build only one element?